internal/dcache/rpc/server/handler.go

```diff
@@ -47,43 +57,894 @@ var _ service.ChunkService = &ChunkServiceHandler{}
+		return nil, rpc.NewResponseError(rpc.ComponentRVsInvalid, fmt.Sprintf("request component RVs %v are invalid for MV %s component RVs %v", req.ComponentRV, req.Chunk.Address.MvName, mvInfo.componentRVs))
+	}
+
+	// check if the sync has started. If yes, block the chunk operations till the sync is completed
+	mvInfo.blockChunkOps()
+
+	// increment the chunk operation count for this MV
+	mvInfo.incrementChunkOps()
+
+	// decrement the chunk operation count for this MV when the function returns
+	defer mvInfo.decrementChunkOps()
+
+	startTime := time.Now()
+
+	// acquire lock for the chunk address to prevent concurrent writes
+	chunkAddress := getChunkAddress(req.Chunk.Address.FileID, req.Chunk.Address.RvID, req.Chunk.Address.MvName, req.Chunk.Address.OffsetInMB)
+	flock := h.locks.Get(chunkAddress)
+	flock.Lock()
+	defer flock.Unlock()
+
+	cacheDir := rvInfo.cacheDir
+	mvSyncState := mvInfo.getSyncStatus()
+
+	var chunkPath, hashPath string
+	var createSyncDir bool
+	if req.IsSync {
+		// sync write RPC call. This is called after the StartSync RPC to copy the contents
+		// from the online RV (lowest index RV) to the new out of sync RV.
+		// In this case the chunks must be written to the regular mv directory, i.e. rv0/mv0
+		if mvSyncState {
+			chunkPath, hashPath = getRegularMVPath(cacheDir, req.Chunk.Address.MvName, req.Chunk.Address.FileID, req.Chunk.Address.OffsetInMB)
+			createSyncDir = true
+		} else {
+			log.Err("ChunkServiceHandler::PutChunk: MV %s is not in sync state, whereas the client request is sync call", req.Chunk.Address.MvName)
+			common.Assert(false, fmt.Sprintf("MV %s is not in sync state, whereas the client request is sync call", req.Chunk.Address.MvName))
+			return nil, rpc.NewResponseError(rpc.InvalidRequest, fmt.Sprintf("MV %s is not in sync state, whereas the client request is sync call", req.Chunk.Address.MvName))
+		}
+	} else {
+		// client write RPC call. If the MV is in sync state, the chunks must be written to the sync directory, i.e. rv0/mv0.sync
+		// If the MV is not in sync state, the chunks must be written to the regular mv directory, i.e. rv0/mv0
+		if mvSyncState {
+			chunkPath, hashPath = getSyncMVPath(cacheDir, req.Chunk.Address.MvName, req.Chunk.Address.FileID, req.Chunk.Address.OffsetInMB)
+			createSyncDir = true
+		} else {
+			chunkPath, hashPath = getRegularMVPath(cacheDir, req.Chunk.Address.MvName, req.Chunk.Address.FileID, req.Chunk.Address.OffsetInMB)
+		}
+	}
+
+	if createSyncDir {
+		//create sync directory if not present
+		syncDir := filepath.Join(cacheDir, req.Chunk.Address.MvName+".sync")
+		err = h.createMVDirectory(syncDir)
+		if err != nil {
+			log.Err("ChunkServiceHandler::PutChunk: Failed to create sync directory %s [%v]", syncDir, err.Error())
+			return nil, rpc.NewResponseError(rpc.InternalServerError, fmt.Sprintf("failed to create sync directory %s [%v]", syncDir, err.Error()))
+		}
+	}
```

Comment:
this is best created in StartSync (and deleted in EndSync, after moving the chunks to the regular mv folder).
If StartSync succeeds, it effectively means that the RV is ready to store chunks from that MV. This includes having a directory to hold them (apart from having enough space)

Reply:
Done, added the sync dir creation and deletion in StartSync and EndSync calls